### PR TITLE
Add swagger support

### PR DESCRIPTION
**Pull request by malacourse**
Please review this request to add swagger support to the spring rest sample.

### CODE REVIEW

**Review by etsauer — CHANGES_REQUESTED** [review]
@malacourse the code looks good, builds and runs properly. However, now when I run the app and hit `localhost:8080/` I get this generic error message saying there's nothing here:

```
Whitelabel Error Page
This application has no explicit mapping for /error, so you are seeing this as a fallback.

Wed Jan 17 16:46:23 EST 2018
There was an unexpected error (type=Not Found, status=404).
No message available
```

Can we put something there? Maybe a mapping of API versions or something more helpful?

**Comment by malacourse** [comment]
I updated code so the default path "/" returns available paths.

**Comment by etsauer** [comment]
@malacourse @sabre1041 I think we should quickly align on version numbers and how they will progress, and how we want to keep them in sync between the API versions and the package version from maven.

Currently the maven `<version/>` is set at `0.1.0` and we haven't changed it since the start. The API version in this PR is `v2`. Since we are aiming to use this app to demonstrate some _real world_ application operations, I would like us to settle on a versioning strategy for this app.

Thoughts?

File: src/main/java/hello/Application.java [hunk 1]
```diff
@@ -4,11 +4,61 @@
...
+            .apiInfo(apiInfo())
+            .select()
+            .paths(apiPaths())
+            .build();
+    }
+    
+
+    private Predicate<String> apiPaths() {
+        return or(
+                regex("/v2/greeting.*"),
```
Comment (etsauer) [inline src/main/java/hello/Application.java]: this file still has several references to v2

**Review by etsauer — CHANGES_REQUESTED** [review]

**Review by etsauer — APPROVED** [review]
LGTM

**Pull request merged**